packages/marketing-modal/src/MarketingModal.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import Button from '@leafygreen-ui/button';
+import { Link } from '@leafygreen-ui/typography';
+import Modal from '@leafygreen-ui/modal';
+import { uiColors } from '@leafygreen-ui/palette';
+
+const titleStyle = css`
+  color: ${uiColors.gray.dark2};
+  font-size: 24px;
+  font-weight: bold;
+  line-height: 25px;
+
+  margin-bottom: 10px;
+`;
+
+const baseModalStyle = css`
+  width: 600px;
+  padding: initial;
+  overflow: scroll;
+`;
+
+const baseCoverStyle = css`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  & > * {
+    display: block;
+  }
+`;
+
+const defaultCoverStyle = css`
+  padding-top: 20px;
+  padding-bottom: 8px;
+`;
+
+const coverCoverStyle = css`
+  padding-bottom: 24px;
+
+  & > * {
+    width: 100%;
+  }
+`;
+
+const contentStyle = css`
+  color: ${uiColors.gray.dark1};
+  font-family: Akzidenz, ‘Helvetica Neue’, Helvetica, Arial, sans-serif;
+  font-size: 16px;
+  letter-spacing: 0;
+  line-height: 24px;
+  text-align: center;
+
+  padding: 0 92px;
+  padding-bottom: 24px;
+`;
+
+const footerContentStyle = css`
+  line-height: 24px;
+  padding-bottom: 50px;
+
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+
+  & > * {
+    padding: 4px;
+  }
+`;
+
+interface MarketingModalProps {
+  title: string;
+  cover: JSX.IntrinsicElements['img'];
+  coverStyle?: 'default' | 'cover';
+  children: React.ReactNode;
+  open?: boolean;
+  className?: string;
+  primaryActionProps: ActionProps;
+  secondaryActionProps: ActionProps;
+}
+
+interface ActionProps {
+  label: string;
+  onClick?: () => void;
+}
+
+const MarketingModal = ({
+  children,
+  title,
+  cover,
+  coverStyle = 'default',
+  primaryActionProps: { label: primaryActionLabel, ...primaryActionProps },
+  secondaryActionProps: {
+    label: secondaryActionLabel,
+    ...secondaryActionProps
+  },
+
+  ...modalProps
+}: MarketingModalProps) => {
+  return (
+    <Modal {...modalProps} contentClassName={baseModalStyle}>
```

Comment:
Clicking the "x" in the top right corner of the modal doesn't force the modal to close, instead the cover image gets highlighted.

Reply:
I actually think that's just because the setOpen function isn't being passed to the Modal; however, when I tried to explicitly pass it, I'm getting a TS error.